ledger/participant-integration-api/src/test/suite/scala/platform/indexer/parallel/ParallelIndexerFactorySpec.scala

```diff
@@ -0,0 +1,311 @@
+    create_argument = None,
+    create_signatories = None,
+    create_observers = None,
+    create_agreement_text = None,
+    create_key_value = None,
+    create_key_hash = None,
+    create_argument_compression = None,
+    create_key_value_compression = None,
+    event_sequential_id = 0,
+  )
+
+  private val someEventExercise = DbDto.EventExercise(
+    consuming = true,
+    event_offset = None,
+    transaction_id = None,
+    ledger_effective_time = None,
+    command_id = None,
+    workflow_id = None,
+    application_id = None,
+    submitters = None,
+    node_index = None,
+    event_id = None,
+    contract_id = "1",
+    template_id = None,
+    flat_event_witnesses = Set.empty,
+    tree_event_witnesses = Set.empty,
+    create_key_value = None,
+    exercise_choice = None,
+    exercise_argument = None,
+    exercise_result = None,
+    exercise_actors = None,
+    exercise_child_event_ids = None,
+    create_key_value_compression = None,
+    exercise_argument_compression = None,
+    exercise_result_compression = None,
+    event_sequential_id = 0,
+  )
+
+  private val someEventDivulgence = DbDto.EventDivulgence(
+    event_offset = None,
+    command_id = None,
+    workflow_id = None,
+    application_id = None,
+    submitters = None,
+    contract_id = "1",
+    template_id = None,
+    tree_event_witnesses = Set.empty,
+    create_argument = None,
+    create_argument_compression = None,
+    event_sequential_id = 0,
+  )
+
+  behavior of "inputMapper"
+
+  it should "provide required Batch in happy path case" in {
+    val actual = ParallelIndexerFactory.inputMapper(
+      metrics = metrics,
+      toDbDto = _ => _ => Iterator(someParty, someParty),
+    )(lc)(
+      List(
+        (
+          (Offset.fromHexString(Ref.HexString.assertFromString("00")), somePackageUploadRejected),
+          1000000001,
+        ),
+        (
+          (
+            Offset.fromHexString(Ref.HexString.assertFromString("01")),
+            somePackageUploadRejected.copy(recordTime =
+              somePackageUploadRejected.recordTime.addMicros(1000)
+            ),
+          ),
+          1000000002,
+        ),
+        (
+          (
+            Offset.fromHexString(Ref.HexString.assertFromString("02")),
+            somePackageUploadRejected.copy(recordTime =
+              somePackageUploadRejected.recordTime.addMicros(2000)
+            ),
+          ),
+          1000000003,
+        ),
+      )
+    )
+    val expected = Batch[Vector[DbDto]](
+      lastOffset = offset("02"),
+      lastSeqEventId = 0,
+      lastRecordTime = someTime.plusMillis(2).toEpochMilli,
+      batch = Vector(
+        someParty,
+        someParty,
+        someParty,
+        someParty,
+        someParty,
+        someParty,
+      ),
+      batchSize = 3,
+      averageStartTime = 1000000001,
+      offsets = Vector("00", "01", "02").map(offset),
+    )
+    actual.copy(batch = Vector.empty) shouldBe expected.copy(batch = Vector.empty)
```

Comment:
Why do you define the value of `expected.batch` above if you discard it here?

Reply:
o damn, because I forgot to test whether the expected has the same entries as the result :man_facepalming: Since this is merged I will make note and fix this in an upcoming PR